backend/inspirehep/alembic/5a0e2405b624_add_citation_type_to_citation_table.py

```diff
@@ -0,0 +1,36 @@
+#
+# This file is part of Invenio.
+# Copyright (C) 2016-2018 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Add citation_type to citation_table"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "5a0e2405b624"
+down_revision = "595c36d68964"
+branch_labels = ()
+depends_on = None
+
+
+def upgrade():
+    """Upgrade database."""
+    op.execute("CREATE TYPE enum_citation_type as ENUM ('citation', 'self_citation')")
```

Comment:
As I remember there was a discussion that self-citation won't be only citation type, so there was a decision that this fiels should be a name of citation type. If it's not valid anymore, I will change it.

Reply:
I have no idea what else you had in mind, and if that's indeed the case, it's better to have multiple columns as flags for the different types, because otherwise you need to have something like an array if a given citation has multiple types.

Reply:
:+1: